src/modules/adaptors/Maker/DSRAdaptor.sol

```diff
@@ -0,0 +1,133 @@
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { DSRManager } from "src/interfaces/external/Maker/DSRManager.sol";
+import { Pot } from "src/interfaces/external/Maker/Pot.sol";
+
+/**
+ * @title DSR Adaptor
+ * @notice Allows Cellars to deposit/withdraw DAI from the DSR.
+ * @author crispymangoes
+ */
+contract DSRAdaptor is BaseAdaptor {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // NOT USED
+    //================= Configuration Data Specification =================
+    // NOT USED
+    //====================================================================
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure override returns (bytes32) {
+        return keccak256(abi.encode("DSR Adaptor V 0.0"));
+    }
+
+    /**
+     * @notice Current networks DSR Manager address.
+     */
+    DSRManager public immutable dsrManager;
+
+    /**
+     * @notice Current networks Pot address.
+     */
+    Pot public immutable pot;
+
+    /**
+     * @notice Current networks DAI address.
+     */
+    ERC20 public immutable dai;
+
+    constructor(address _dsrManager) {
+        dsrManager = DSRManager(_dsrManager);
+        pot = Pot(dsrManager.pot());
+        dai = ERC20(dsrManager.dai());
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice Deposit assets directly into DSR.
+     */
+    function deposit(uint256 assets, bytes memory, bytes memory) public override {
+        _join(assets);
+    }
+
+    /**
+     * @notice Withdraw assets from DSR.
+     */
+    function withdraw(uint256 assets, address receiver, bytes memory, bytes memory) public override {
+        _externalReceiverCheck(receiver);
+
+        dsrManager.exit(receiver, assets);
+    }
+
+    /**
+     * @notice Identical to `balanceOf`.
+     * @dev Does not account for pending interest.
+     */
+    function withdrawableFrom(bytes memory, bytes memory) public view override returns (uint256) {
+        uint256 pieOf = dsrManager.pieOf(msg.sender);
+        return pieOf.mulDivDown(pot.chi(), 1e27);
+    }
+
+    /**
+     * @notice Returns the balance of DAI in the DSR.
+     * @dev Does not account for pending interest.
+     */
+    function balanceOf(bytes memory) public view override returns (uint256) {
+        uint256 pieOf = dsrManager.pieOf(msg.sender);
+        return pieOf.mulDivDown(pot.chi(), 1e27);
```

Comment:
Is this okay to not account for pending interest? `chi` is updated upon `joins` and `exits` via [pot.drip](https://etherscan.deth.net/address/0x373238337Bfe1146fb49989fc222523f83081dDb) iirc.

Or are we not accounting for interest / earnings to be conservative?